### DISCUSSION TITLE
Filtering for Custom Field

### DISCUSSION THREAD

**Discussion by RobinBeismann** [netbox-community/netbox#7907] [Q&A]
Hey,

I'm trying to filter on the REST API using a custom field, but I can't get it to work..
Basically, I want all devices where "monitoring_enabled" equals either null (dropdown not yet set) or "false", but filtering for neither works..
I've tried the following to filter for false:
`/api/virtualization/virtual-machines/?cf_monitoring_enabled=false`

and this to filter for empty values:
`/api/virtualization/virtual-machines/?cf_monitoring_enabled__empty=true`

However both of them just return all virtual machines.. 
I'm using v3.0.10.

This is how my custom fields look in the API Response:
```
 "custom_fields": {
                "monitoring_enabled": true,
                "monitoring_autodiscover": false,
                "monitoring_group": null,
                "monitoring_template": null
            }
```

**Comment by MalfuncEddie** [comment]
Did you ever find a solution for this?

**Reply by RobinBeismann** [reply]
Yeah, my filter was simply wrong, the __empty=true is invalid. Which query are you trying to run? Maybe I can help.

**Reply by MalfuncEddie** [reply]
Basically the same as your example. Trying to filter out vms that have custom fields that are null or "" 

So I would guess __empty=false would be my case but it does not work. I do see some issues that are resolved with the empty filter in te last version so I'll have to update I guess .

**Reply by candlerb** [reply]
I don't think custom fields support [lookup expressions](https://demo.netbox.dev/static/docs/rest-api/filtering/#lookup-expressions).  The documentation says:

> **Certain model fields** also support filtering using additional lookup expressions

(My emphasis).

A quick search suggests this was raised already as #8821, but the ever-vigilant bot has auto-closed it.

It's also unfortunate that if you put a query parameter like `cf_foo__empty` but this syntax is invalid, it's silently accepted and ignored, rather than giving a helpful error.  That's why you see all VMs.

**Reply by MalfuncEddie** [reply]
No problem, I think I can just solve this with a custom script or report as work around. I was just wondering if it is a bug/not implemented feature. If it is only on Certain model fields  and works as designed I got no issue with that :)
